src/util/util.c

```diff
@@ -379,14 +379,19 @@ void free_tempfile(struct tempfile *tf) {
 struct tempfile *make_tempfile(void) {
    int fd;
    struct tempfile *tf;
+   char *tmpdir = singularity_registry_get("TMPDIR");
+
+   if (tmpdir == NULL) {
+       tmpdir = "/tmp";
```

Comment:
Re ` hard coded /tmp` -- shouldn't regular, not the prefixed one from singularity_registry_get, TMPDIR env be consulted for that ideally here?

Reply:
I tried to use just `TMPDIR`. I'm thinking it's getting lost somewhere as doing `getenv("TMPDIR");` returned NULL.

So, I went with the next best thing. Haven't had a chance for one of the guys who knows the code better than I to look over it yet, but if they have an idea on how to use just the `TMPDIR` value, I'd prefer that over using `SINGULARITY_TMPDIR`.

Reply:
yeah, and here anyways `/tmp` was hardcoded already, so this fix doesn't have to fix that